backend/pkg/metrics/prometheus_metrics.go

```diff
@@ -0,0 +1,158 @@
+package metrics
+
+import (
+	"fmt"
+	"github.com/go-openapi/strfmt"
+	"github.com/openclarity/kubeclarity/api/server/restapi/operations"
+	"github.com/openclarity/kubeclarity/backend/pkg/database"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"strings"
+	"time"
+)
+
+const (
+	// Prefix is to make metrics unique
+	Prefix string = "kubeclarity"
+)
+
+var (
+	applicationCounter = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: Prefix + "_number_of_applications",
+		Help: "The total number of applications",
+	})
+	resourceCounter = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: Prefix + "_number_of_resources",
+		Help: "The total number of resources",
+	})
+	packageCounter = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: Prefix + "_number_of_packages",
+		Help: "The total number of packages",
+	})
+	vulnerabilityCounter = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: Prefix + "_number_of_vulnerabilities_total",
+		Help: "The total number of vulnerabilities",
+	})
+	fixableVulnerabilityCounter = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: Prefix + "_number_of_fixable_vulnerabilities_total",
+		Help: "The total number of fixable vulnerabilities",
+	})
+	fixableVulnerability = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: Prefix + "_number_of_fixable_vulnerabilities",
+		Help: "The number of fixable vulnerabilities per severity",
+	}, []string{"severity"})
+	vulnerability = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: Prefix + "_number_of_vulnerabilities",
+		Help: "The number of vulnerabilities per severity",
+	}, []string{"severity"})
+	trendGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: Prefix + "_vulnerability_trend",
+		Help: "Incoming vulnerabilities within the last hour",
+	}, []string{"severity"})
+	applicationGauge = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: Prefix + "_application_vulnerability",
+		Help: "Count of vulnerabilities per application and level",
+	}, []string{"name", "environment", "severity"})
+)
+
+type Server struct {
+	refreshInterval int
+	dbHandler       *database.Handler
+}
+
+func ProduceMetrics(dbHandler *database.Handler, refreshInterval int) *Server {
+	return &Server{
+		refreshInterval: refreshInterval,
+		dbHandler:       dbHandler,
+	}
+}
+
+func init() {
+	logrus.Infof("Adding /metrics endpoint to http server assumed started with healthz")
+	http.Handle("/metrics", promhttp.Handler())
```

Comment:
would it make sense to avoid using `init()` and move the code under `ProduceMetrics`?